cpp/src/parquet/bloom_filter.cc

```diff
@@ -65,51 +69,109 @@ void BlockSplitBloomFilter::Init(const uint8_t* bitset, uint32_t num_bytes) {
   PARQUET_ASSIGN_OR_THROW(data_, ::arrow::AllocateBuffer(num_bytes_, pool_));
   memcpy(data_->mutable_data(), bitset, num_bytes_);
 
-  this->hasher_.reset(new MurmurHash3());
+  this->hasher_ = std::make_unique<XxHasher>();
 }
 
-BlockSplitBloomFilter BlockSplitBloomFilter::Deserialize(ArrowInputStream* input) {
-  uint32_t len, hash, algorithm;
-  int64_t bytes_available;
+static constexpr uint32_t kBloomFilterHeaderSizeGuess = 32;
+static constexpr uint32_t kMaxBloomFilterHeaderSize = 1024;
```

Comment:
I think this is pointless complication. We can try to read up to 256 bytes at once, which should be sufficient for future additions.

Reply:
I think 32B should be enough for a long time...

Reply:
What I mean is that more generally we can read a fixed size and simply error out if deserializing fails.

Reply:
Emmmm, I got what you mean...I want use `Read` instead of `Peek` for compability, and can only `Read` once to make the program simplier...

Reply:
Yes, exactly ;-)